decoder.go

```diff
@@ -73,7 +73,7 @@ func convertSignedInts(src interface{}, dst *reflect.Value) error {
 			return nil
 		}
 	case uintptr:
-		if t <= math.MaxInt64 && !dst.OverflowInt(int64(t)) {
+		if int64(t) <= math.MaxInt64 && !dst.OverflowInt(int64(t)) {
```

Comment:
If you cast t to int64 it will necessarily be <= math.MaxInt64.  So the comparison `int64(t) <= math.MaxInt64` will always be true right?  I think what you need to do here is cast to an unsigned type and then do your comparison.  Maybe something like:

    if uint64(t) <= math.MaxInt64 ...

Also, just a question.  Why, in the first case block in this switch, do you convert the integer value 't' to a string and then parse it again?  Can't you just cast 't' to int64, check for overflow and then call SetInt()?

And why aren't you checking for overflow in the uint64 case?

Reply:
I am using Print-Parse pair, because in case of multiple types in the type switch, `t` is going to be an `interface{}` and I can't use it to call `Value.SetInt`.
So either this or have multiple small `case`s for each int8, 16, ...

Reply:
Ah.  I'm unfamiliar with the nuances of type assertion switches.  I think you should think about  going with the small case statements and avoid running the src value through print/parse.

Reply:
Well, that was my initial implementation :) I've received a comment about bad readability and put all integer numbers together. We already doing lots of reflection, which always allocates on heap, so there is not much perf gain we can receive.

Reply:
Never be afraid to reject bad advice.  I think it's good to always try to think of your code and write your code as if it was an example of what to do for other engineers and assume they will copy what you do.  In other words, everything you write is a teaching opportunity.  I don't think you can say that about this print/parse integer code.  Some engineer will come along and see this and say "Hey, that's a neat trick, I should use that", without realizing the implications.

Reply:
I am happy to chat about these improvements offline since it feels like an off topic :)